ma/utility.py

```diff
@@ -2,23 +2,26 @@
 
 def get_pretty_print(results):
     return json.dumps(
-            results, 
+            results,
             sort_keys=True,
-            indent=4, 
+            indent=4,
             separators=(',', ': '))
 
 def pretty_print(results):
     print(get_pretty_print(results))
 
 def get_dict_from_named_tuple(nt):
     a = dir(nt)
-    d = { 
-            k: getattr(nt, k) 
-            for k 
-            in a 
+    d = {
+            k: getattr(nt, k)
+            for k
+            in a
             if k[0] != '_' \
                 and k != 'count' \
-                and k != 'index' 
+                and k != 'index'
         }
 
     return d
+
+def match_dict_keys_from_list(d, l):
+    return [(k, d[k]) for k in l if k in d]
```

Comment:
Interesting.

My heart always flutters when I see a nice, tight list-comprehension that uses an `if`. I think constructs like that are what slid the ring onto my finger with Python even if mission-critical code that requires concurrency has to be written in C or other services.